### PR TITLE
Add truly async primer dialogs using custom turbo_stream action

### PR DESCRIPTION
**Pull request by oliverguenther**
Replace our current implementation of the `AsyncDialogComponent` which still needed to render parts of the dialog in DOM with a custom turbo_stream action that will render the dialog dynamically, and also remove it again once it's closed.

### CODE REVIEW

File: modules/overviews/app/components/project_custom_fields/sections/edit_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<%=
+  component_wrapper do
+    primer_form_with(
+      model: @project,
+      method: :put,
+      data: { turbo: true, turbo_stream: true, "test-selector": "async-dialog-content" },
+      url: update_project_custom_values_path(project_id: @project.id, section_id: @project_custom_field_section.id),
+    ) do |f|
+      component_collection do |collection|
+        collection.with_component(Primer::Alpha::Dialog::Body.new(my: 3, classes: "Overlay-body_autocomplete_height")) do
```
Comment (HDinger) [inline modules/overviews/app/components/project_custom_fields/sections/edit_component.html.erb]: We do not need the `Dialog::Body` here any more, as it is now correctly set by the Dialog. Currently, this results in additional outer spacing:

Bildschirmfoto 2024-04-18 um 11 20 09

File: modules/overviews/app/components/project_custom_fields/sections/edit_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          render(Projects::CustomFields::Form.new(
+            f,
+            project: @project,
+            custom_field_section: @project_custom_field_section,
+            wrapper_id: "#edit-project-custom-fields-dialog-#{@project_custom_field_section.id}"
+          ))
+        end
+        collection.with_component(Primer::Alpha::Dialog::Footer.new) do
```
Comment (HDinger) [inline modules/overviews/app/components/project_custom_fields/sections/edit_component.html.erb]: Same as above. If we can use the footer correctly, we'll also benefit from the correct scrolling behavior on longer modals. Currently, we still have to scroll the whole content to see the "Save" and "Cancel" buttons.

**Review by HDinger — CHANGES_REQUESTED** [review]
Great job, @oliverguenther ! 🥳 I really like the new approach, it reads so much easier and should also be much more intuitive to use 👍 I only have a small remark on the project attribute dialog. Otherwise, this is good to go from my side.

**Review by HDinger — APPROVED** [review]
Good to merge once the ci is green 👍

**Comment by oliverguenther** [comment]
Remaining spec is the calendar widget spec flickering

**Pull request merged**